File: fizzbuzz/python/Arnold/fizzbuzz_test.py
import fizzbuzz


def validate_file(output, lines):
    file = open("output.txt", "r")
    output = output.split("\n")
    i = 0
    while i < lines:
        line = file.readline()[:-1]
        assert output[i] == line
        i += 1


def test_fizz():
    output = fizzbuzz.fizz_buzz(3)
    validate_file(output, 3)


def test_buzz():
    output = fizzbuzz.fizz_buzz(5)
    validate_file(output, 5)


def test_none():
    output = fizzbuzz.fizz_buzz(0)
    assert output == "Invalid number. Please input number greater than 0."


def test_fizz_buzz():
    output = fizzbuzz.fizz_buzz(15)
    validate_file(output, 15)


def test_ping():
    output = fizzbuzz.fizz_buzz(7)
    validate_file(output, 7)


def test_fizz_ping():
    output = fizzbuzz.fizz_buzz(21)
    validate_file(output, 21)


def test_buzz_ping():
    output = fizzbuzz.fizz_buzz(35)
    validate_file(output, 35)


def test_fizz_buzz_ping():
    output = fizzbuzz.fizz_buzz(105)
    validate_file(output, 105)


def test_pong():
    output = fizzbuzz.fizz_buzz(11)
    validate_file(output, 11)


def test_fizz_pong():
    output = fizzbuzz.fizz_buzz(33)
    validate_file(output, 33)


def test_buzz_pong():
    output = fizzbuzz.fizz_buzz(55)
    validate_file(output, 55)


def test_fizz_buzz_pong():
    output = fizzbuzz.fizz_buzz(165)
    validate_file(output, 165)


def test_ping_pong():
    output = fizzbuzz.fizz_buzz(77)
    validate_file(output, 77)


def test_fizz_ping_pong():
    output = fizzbuzz.fizz_buzz(231)
    validate_file(output, 231)

def test_buzz_ping_pong():
    output = fizzbuzz.fizz_buzz(385)
    validate_file(output, 385)


def test_fizz_buzz_ping_pong():
    output = fizzbuzz.fizz_buzz(1155)
    validate_file(output, 1155)

File: fizzbuzz/Go/alexis/fizzbuzz_test.go
package main

import (
	"testing"
)

func TestFizz( t *testing.T )  {
	result := fizzbuzz(3)
	if result != "fizz" {
		t.Errorf("testFizz failed, expected: fizz, got %v", result)
	} else {
		t.Logf("testFizz success, expected: fizz, got %v", result)
	}
}

func TestBuzz( t *testing.T )  {
	result := fizzbuzz(5)
	if result != "buzz" {
		t.Errorf("testFizz failed, expected: buzz, got %v", result)
	} else {
		t.Logf("testFizz success, expected: buzz, got %v", result)
	}
}

func TestPing( t *testing.T )  {
	result := fizzbuzz(7)
	if result != "ping" {
		t.Errorf("testFizz failed, expected: ping, got %v", result)
	} else {
		t.Logf("testFizz success, expected: ping, got %v", result)
	}
}

func TestPong( t *testing.T )  {
	result := fizzbuzz(11)
	if result != "pong" {
		t.Errorf("testFizz failed, expected: pong, got %v", result)
	} else {
		t.Logf("testFizz success, expected: pong, got %v", result)
	}
}

func TestFizzBuzz( t *testing.T )  {
	result := fizzbuzz(15)
	if result != "fizzbuzz" {
		t.Errorf("testFizz failed, expected: fizzbuzz, got %v", result)
	} else {
		t.Logf("testFizz success, expected: fizzbuzz, got %v", result)
	}
}

func TestFizzPing( t *testing.T )  {
	result := fizzbuzz(21)
	if result != "fizzping" {
		t.Errorf("testFizz failed, expected: fizzping, got %v", result)
	} else {
		t.Logf("testFizz success, expected: fizzping, got %v", result)
	}
}

func TestFizzPong( t *testing.T )  {
	result := fizzbuzz(33)
	if result != "fizzpong" {
		t.Errorf("testFizz failed, expected: fizzpong, got %v", result)
	} else {
		t.Logf("testFizz success, expected: fizzpong, got %v", result)
	}
}

func TestFizzBuzzPing( t *testing.T )  {
	result := fizzbuzz(105)
	if result != "fizzbuzzping" {
		t.Errorf("testFizz failed, expected: fizzbuzzping, got %v", result)
	} else {
		t.Logf("testFizz success, expected: fizzbuzzping, got %v", result)
	}
}

func TestFizzBuzzPong( t *testing.T )  {
	result := fizzbuzz(165)
	if result != "fizzbuzzpong" {
		t.Errorf("testFizz failed, expected: fizzbuzzpong, got %v", result)
	} else {
		t.Logf("testFizz success, expected: fizzbuzzpong, got %v", result)
	}
}

func TestFizzBuzzPingPong( t *testing.T )  {
	result := fizzbuzz(1155)
	if result != "fizzbuzzpingpong" {
		t.Errorf("testFizz failed, expected: fizzbuzzpingpong, got %v", result)
	} else {
		t.Logf("testFizz success, expected: fizzbuzzpingpong, got %v", result)
	}
}

func TestBuzzPing( t *testing.T )  {
	result := fizzbuzz(35)
	if result != "buzzping" {
		t.Errorf("testFizz failed, expected: buzzping, got %v", result)
	} else {
		t.Logf("testFizz success, expected: buzzping, got %v", result)
	}
}

func TestBuzzPong( t *testing.T )  {
	result := fizzbuzz(55)
	if result != "buzzpong" {
		t.Errorf("testFizz failed, expected: buzzpong, got %v", result)
	} else {
		t.Logf("testFizz success, expected: buzzpong, got %v", result)
	}
}

func TestBuzzPingPong( t *testing.T )  {
	result := fizzbuzz(385)
	if result != "buzzpingpong" {
		t.Errorf("testFizz failed, expected: buzzpingpong, got %v", result)
	} else {
		t.Logf("testFizz success, expected: buzzpingpong, got %v", result)
	}
}

func TestPingPong( t *testing.T )  {
	result := fizzbuzz(77)
	if result != "pingpong" {
		t.Errorf("testFizz failed, expected: pingpong, got %v", result)
	} else {
		t.Logf("testFizz success, expected: pingpong, got %v", result)
	}
}
